### ISSUE TITLE
Could not read source map for localhost source with Unexpected 503 response from source map at local 127.0.0.1 IP address

### ISSUE THREAD

**Issue by mkormendy** [microsoft/vscode-js-debug#1336]
**Describe the bug**
When using Ionic 5, Angular 13 and and JS Debug extension, I'm getting many errors like the following when attempting to debug the app:

```
Could not read source map for http://localhost:8100/styles.js: Unexpected 503 response from http://127.0.0.1:8100/styles.js.map: connect ECONNREFUSED 127.0.0.1:8100
```
What's interesting is that loading up the source map address with localhost `http://localhost:8100/styles.js.map` instead of the IP address version `http://127.0.0.1:8100/styles.js.map` retrieves the file and doesn't have a 503 response.

**VS Code Version:** 1.69.2 on MacOS Monterey on Silicon M1

**Comment by mkormendy** [comment]
What's ALSO interesting, if I change the `devServerAddress` in the `launch.json` to 127.0.0.1, then the errors go away.

**Comment by connor4312** [comment]
Thanks for the issue. I assume your launch config opens `127.0.0.1:4200` or similar as its URL?

macOS does a lot of spooky things with ipv6. For me, if I'm on a network with ipv6, then localhost binds to `::1` by default, as it appeared to do for you. However, in that case running `ng serve` which binds to localhost isn't accessible at `127.0.0.1:4200` at all (though in the past I've seen this sometimes work in some applications...) I would also guess that changing your `url` to point at `localhost` instead of `127.0.0.1` would make this work.

**Comment by lowickert** [comment]
I'm getting the same error on windows 10 when using vs code and the js debugger using Angular 14. I can also access the source file from the localhost-url but not the ip-url. The url in the launch.json is pointing to "http://localhost:4200", that is not where the error is. 
I haven't yet found a fix for that. Could it be that the error is in the js debugger?

**Comment by connor4312** [comment]
localhost is not always the same as `127.0.0.1` when on ipv6-enabled networks

**Comment by tulipandmills** [comment]
It took me weeks to end up on this solution. I think it is very important to get this fixed for a clean installed Mac Os with the latest Angular Cli. ng new and vs code debug should be working out of the box

**Comment by zosrothko** [comment]
From my side, solved by replacing localhost by 127.0.0.1 as
`"url": "http://127.0.0.1:4200/"`
